crates/worker/src/partition/mod.rs

```diff
@@ -448,6 +455,48 @@ where
                     .await
                     .context("failed handling StepDown command")?;
             }
+            PartitionProcessorControlCommand::CreateSnapshot(maybe_sender) => {
+                if let Some(inflight_snapshot) = self.inflight_create_snapshot_task.take() {
+                    if !inflight_snapshot.is_finished() {
```

Comment:
I'm not in love with how this came out but I couldn't make it better - suggestions welcome!

Reply:
Aha!

```
if self
    .inflight_create_snapshot_task
    .as_ref()
    .is_some_and(|task| !task.is_finished())
{
    ...
}
```